File: front/vue/src/api/friend.js
import api from '@/api'

// 根据id查询好友信息
export function getFiendInfo(id){
    return api.get('/friend/info',{
        params: {
            id: id
        }
    })
}

// 获取好友列表
export function getFriendList(id){
    return api.get('/friend/list',{
        params: {
            id: id
        }
    })
}

// 根据邮箱获取用户信息
export function searchUserInfoByEmail(email){
    return api.get('/friend/search',{
        params: {
            email: email
        }
    })
}

File: back/routers/friend.go
package routers

import (
	"back/configs"
	"back/controllers"
	"back/repository"
	"back/service"
	"github.com/gin-gonic/gin"
)

func SetFriendRoutes(r *gin.Engine) {
	// 初始化依赖
	authRepository := repository.NewAuthRepository(configs.MysqlDb)
	friendRepository := repository.NewFriendRepository(configs.MysqlDb)
	messageRepository := repository.NewMessageRepository(configs.MysqlDb)

	friendService := service.NewFriendService(authRepository, friendRepository, messageRepository)

	friendController := controllers.NewFriendController(friendService)

	// 配置路由
	friend := r.Group("/friend")
	{
		// 根据id查询好友详情
		friend.GET("/info", friendController.GetFriendInfo)
		// 获取好友列表
		friend.GET("/list", friendController.GetFriendList)
		// 根据邮箱查询用户信息，用于添加好友
		friend.GET("/search", friendController.SearchUserByEmail)
		// 添加好友请求
		friend.POST("/add", friendController.AddFriend)
		// 删除好友
		friend.DELETE("/delete", friendController.DeleteFriend)
	}
}
